src/ucp/am/eager_multi.c

```diff
@@ -209,32 +209,31 @@ static UCS_F_ALWAYS_INLINE ucs_status_t ucp_proto_am_eager_multi_zcopy_send_func
         ucp_am_mid_hdr_t middle;
     } hdr;
     ucp_am_id_t am_id;
-    size_t hdr_size, meta_size, footer_size, footer_offset, iov_count;
+    size_t hdr_size, footer_size, footer_offset, iov_count;
+    ucp_am_first_ftr_t *ftr;
     uct_iov_t iov[UCP_MAX_IOV];
 
     if (req->send.state.dt_iter.offset == 0) {
         am_id         = UCP_AM_ID_AM_FIRST;
         hdr_size      = sizeof(hdr.first);
-        meta_size     = UCP_AM_FIRST_FRAG_META_LEN + user_hdr_size;
-        footer_size   = sizeof(ucp_am_first_ftr_t) + user_hdr_size;
+        footer_size   = sizeof(*ftr) + user_hdr_size;
         footer_offset = 0;
         ucp_am_fill_header(&hdr.first, req);
         /* The method also fills middle/last fragment footer. The footer can be
            reused by all fragments because it is immutable. */
-        ucp_am_fill_first_footer(UCS_PTR_BYTE_OFFSET(
-                req->send.msg_proto.am.reg_desc + 1, user_hdr_size), req);
+        ftr = UCS_PTR_BYTE_OFFSET(req->send.msg_proto.am.reg_desc + 1,
+                                  user_hdr_size);
+        ucp_am_fill_first_footer(ftr, req);
     } else {
         am_id         = UCP_AM_ID_AM_MIDDLE;
         hdr_size      = sizeof(hdr.middle);
-        meta_size     = UCP_AM_MID_FRAG_META_LEN;
-        footer_size   = sizeof(ucp_am_mid_ftr_t);
+        footer_size   = sizeof(ftr->super);
         footer_offset = user_hdr_size;
         ucp_am_fill_middle_header(&hdr.middle, req);
     }
 
-    iov_count = ucp_proto_am_eager_multi_zcopy_add_payload(req, lpriv,
-                                                           meta_size, next_iter,
-                                                           iov);
+    iov_count = ucp_proto_am_eager_multi_zcopy_add_payload(
+            req, lpriv, sizeof(ucp_am_hdr_t) + footer_size, next_iter, iov);
```

Comment:
since we use sizeof(ucp_am_hdr_t) and not "hdr_size" here:
it means we can `STATIC_ASSERT(sizeof(hdr.first)==sizeof(ucp_am_hdr_t))` and `STATIC_ASSERT(sizeof(hdr.middle)==sizeof(ucp_am_hdr_t))`, and then remove hdr_size variable and replace it by sizeof(ucp_am_hdr_t)